### ISSUE TITLE
site-replicator-0 service account not working for backup site

### ISSUE THREAD

**Issue by pschichtel** [minio/minio#19217]
If updated to the latest release that includes @poornas' PR that fixes #19206. Service accounts indeed started working again, however my backup site still complains, that the production site is offline because the site-replicator-0 access key seems wrong:

```
minio-1  | API: SRPeerBucketOps(bucket=bucket-name)
minio-1  | Time: 12:13:23 UTC 03/07/2024
minio-1  | DeploymentID: bc54da3b-88f4-4a0d-a9d4-2365bf5a0d80
minio-1  | RequestID: 17BA7A69D7884F87
minio-1  | RemoteHost: ...
minio-1  | Host: backup-site
minio-1  | UserAgent: MinIO (linux; amd64) madmin-go/2.0.0
minio-1  | Error: Site replication error(s): 
minio-1  | 'ConfigureReplication' on site Production (bucket-name): failed(Backup->Production: Bucket target creation error: Remote service endpoint offline, target bucket: bucket-name or remote service credentials: site-replicator-0 invalid 
minio-1  | 	The Access Key Id you provided does not exist in our records.) (*errors.errorString)
minio-1  |        3: internal/logger/logger.go:260:logger.LogIf()
minio-1  |        2: cmd/admin-handlers-site-replication.go:143:cmd.adminAPIHandlers.SRPeerBucketOps()
minio-1  |        1: net/http/server.go:2136:http.HandlerFunc.ServeHTTP()
```

## Your Environment

* Version used (`minio --version`): RELEASE.2024-03-07T00-43-48Z
* Server setup and configuration: deployed by operator (5.0.12), replicating to a single-node setup on the same version deployed with docker-compose.
* Operating System and version (`uname -a`): `Linux  6.1.0-18-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.76-1 (2024-02-01) x86_64 GNU/Linux`

**Comment by poornas** [comment]
Did you upgrade both the sites to latest release? Are they both using same root user here

**Comment by pschichtel** [comment]
they are both on latest, yes. the root user has the same name, but different passwords.

**Comment by poornas** [comment]
are you able to do `mc admin user svcacct ls <siteB> <root-user>` and see the site-replicator-0 account? it seems like the iam system was not initialized when this request was sent

**Comment by pschichtel** [comment]
@poornas yep, I see it. this request comes repeatedly in the siteB, so after startup or so.

**Comment by pschichtel** [comment]
@poornas Login works again now, but reloading the console (F5) now leads to logout with "Invalid session" / "The Access Key Id you provided does not exist in our records." on `/session`. The same happens on `/service-accounts` on 1 out of 10 tries or so. This only happens on the production site, the backup site's console seems to work fine. The backup site still very occasionally logs the above error. I assume the symptoms (occasional /service-accounts fails and backup site occasionally unable to authenticate at production site) are related.

Should I create a new issue?

**Comment by harshavardhana** [comment]
This may be related to some notification problem between your nodes. There may be a disconnection between the servers that are "logged," and it cannot update all servers. Causing 1/10 fails depends on which server the F5 routes to. 

Perhaps one of the pods is not up or misbehaving.

**Comment by pschichtel** [comment]
@harshavardhana they are all up and don't seem to have an obvious issue. Network is fine between the nodes, I checked that already. There are also no other node issues that I'm aware of (high CPU, memory or disk pressure).

**Comment by harshavardhana** [comment]
@pschichtel you are sure all nodes have been upgraded?

**Comment by pschichtel** [comment]
I checked the pods one by one. They all showed the correct image name. Then I did a restart on the STS (`kubectl rollout restart sts/...`), but that only restarted the last pod for some reason. Then I manually deleted the pods one by one, while waiting for all drives to be online between pods. The redeployments of pods 1 and 0 caused the console to be partially unavailable , but after all have been restarted I was unable to reproduce the issue so far. I'll keep monitoring the situation.
